js/src/modals/Shapeshift/AwaitingDepositStep/awaitingDepositStep.spec.js

```diff
@@ -48,4 +52,61 @@ describe('modals/Shapeshift/AwaitingDepositStep', () => {
     render({ depositAddress: 'xyz' });
     expect(component.find('FormattedMessage').first().props().id).to.match(/awaitingDeposit/);
   });
+
+  describe('instance methods', () => {
+    describe('renderAddress', () => {
+      let address;
+
+      beforeEach(() => {
+        address = shallow(instance.renderAddress(TEST_ADDRESS));
+      });
+
+      it('renders the address', () => {
+        expect(address.text()).to.contain(TEST_ADDRESS);
+      });
+
+      describe('CopyToClipboard', () => {
+        let copy;
+
+        beforeEach(() => {
+          copy = address.find('Connect(CopyToClipboard)');
+        });
+
+        it('renders the copy', () => {
+          expect(copy.length).to.equal(1);
+        });
+
+        it('passes the address', () => {
+          expect(copy.props().data).to.equal(TEST_ADDRESS);
+        });
+      });
+
+      describe('QrCode', () => {
+        let qr;
+
+        beforeEach(() => {
+          qr = address.find('QrCode');
+        });
+
+        it('renders the QrCode', () => {
+          expect(qr.length).to.equal(1);
+        });
+
+        it('passed the address', () => {
+          expect(qr.props().value).to.equal(TEST_ADDRESS);
+        });
+
+        describe('protocol link', () => {
```

Comment:
The link wraps the QrCode. Not attached to it either way, could be outside as well.